driver/src/main/java/org/neo4j/driver/internal/InternalStatementResult.java

```diff
@@ -319,6 +321,7 @@ private boolean tryFetchNext()
         {
             if ( done )
             {
+                connectionHandler.resultBuffered();
```

Comment:
if you do `connectionHandler.resultBuffered()` at the place where `done` is set to be true, then you probably could also address failure/ignore cases too.